src/repr/src/explain/tracing.rs

```diff
@@ -45,8 +45,10 @@ pub struct TraceEntry<T> {
     /// Used to impose global sorting when merging multiple `TraceEntry`
     /// arrays in a single array.
     pub instant: std::time::Instant,
-    /// The time it took to run this optimization step.
-    pub duration: std::time::Duration,
+    /// The duration since the start of the enclosing span.
+    pub span_duration: std::time::Duration,
+    /// The duration since the start of the top-level span seen by the `PlanTrace`.
+    pub full_duration: std::time::Duration,
```

Comment:
Some context why the new field is needed:

- The `OptimizerTrace` maintains a list of traces.
- Each `TraceEntry` is identified by a path `foo/bar/baz` which represents the stack of spans in which the trace was created.
- The `duration` field (now called `span_duration`) was measuring the time between the creation of the last entry in that path (that is, the closest span in the stack) and the time the `trace` call was created. This is useful if you want to measure the time of individual transformations.
- The `full_duration` field will now measure the time between the top-level entry in the stack and the time the `trace` call was created. Assuming the top-level entry is shared for the entire compilation pipeline (which it is, look for `span!(Level::INFO, "optimize")` in the code), this will now correspond to optimization time until the corresponding entry.